Source/LinqToDB/Linq/SequentialAccessHelper.cs

```diff
@@ -0,0 +1,301 @@
+﻿using System;
+using System.Data;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace LinqToDB.Linq
+{
+	using Extensions;
+	using LinqToDB.Expressions;
+	using LinqToDB.Linq.Internal;
+	using LinqToDB.Reflection;
+
+	internal static class SequentialAccessHelper
+	{
+		public static Expression OptimizeMapingExpressionForSequentialAccess(Expression expression, int fieldCount, bool reduce)
+		{
+			if (reduce)
+				expression = expression.Transform(e =>
+				{
+					if (e is ConvertFromDataReaderExpression conv)
+						return conv.Reduce();
+
+					return e;
+				});
```

Comment:
optional. I think it reads better, but that's just me
```suggestion
				expression = expression.Transform(e =>
					e is ConvertFromDataReaderExpression conv
						? conv.Reduce()
						: e;
```